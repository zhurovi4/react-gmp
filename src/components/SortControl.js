import React, { useState } from "react";
import styles from "./SortControl.module.css";

const SortControl = ({ defaultSelection }) => {
  const [currentSelection, setCurrentSelection] = useState(defaultSelection);

  const handleSelectionChange = (event) => {
    const newValue = event.target.value;
    setCurrentSelection(newValue);
  };

  return (
    <div className={styles.sortControl}>
      <label htmlFor="sort-control">Sort by:</label>
      <select
        id="sort-control"
        value={currentSelection}
        onChange={handleSelectionChange}
      >
        <option value="release-date">Release Date</option>
        <option value="title">Title</option>
      </select>
    </div>
  );
};

export default SortControl;
