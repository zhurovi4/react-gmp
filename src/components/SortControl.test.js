import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SortControl from "./SortControl";

describe("SortControl", () => {
  test("matches snapshot", () => {
    const { container } = render(
      <SortControl currentSelection="release-date" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders label and select elements", () => {
    render(<SortControl currentSelection="release-date" />);

    const labelElement = screen.getByLabelText("Sort by:");
    const selectElement = screen.getByRole("combobox", { name: "Sort by:" });

    expect(labelElement).toBeInTheDocument();
    expect(selectElement).toBeInTheDocument();
  });
});
