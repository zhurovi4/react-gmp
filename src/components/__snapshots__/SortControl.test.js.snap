// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortControl matches snapshot 1`] = `
<div>
  <div
    class="sortControl"
  >
    <label
      for="sort-control"
    >
      Sort by:
    </label>
    <select
      id="sort-control"
    >
      <option
        value="release-date"
      >
        Release Date
      </option>
      <option
        value="title"
      >
        Title
      </option>
    </select>
  </div>
</div>
`;
